lib/gooddata/lcm/actions/synchronize_ldm.rb

```diff
@@ -35,57 +35,79 @@ class SynchronizeLdm < BaseAction
 
         description 'Allows to have facts with higher precision decimals'
         param :exclude_fact_rule, instance_of(Type::BooleanType), required: false, default: false
+
+        description 'Specifies how to synchronize LDM and resolve possible conflicts'
+        param :synchronize_ldm, instance_of(Type::SynchronizeLDM), required: false, default: 'diff_against_master_with_fallback'
       end
 
       class << self
         def call(params)
-          include_ca = params.include_computed_attributes.to_b
           exclude_fact_rule = params.exclude_fact_rule.to_b
-
+          client = params.gdc_gd_client
           results = []
+          synchronize = params.synchronize.map do |segment_info|
+            sync_segment(client, exclude_fact_rule, params, results, segment_info)
```

Comment:
since we already pass params, which contain all of the passed variables, why not just pass params and segment_info and unpack it inside sync_segment ?

Reply:
also the passing of results does not really make much sense in this abstraction

Reply:
What do you mean by "does not really make much sense in this abstraction"?

Reply:
We are creating an array of results, which we are passing by reference to the method sync_segment and relying on the side effect of the method on providing data into the results array, which is not really transparent. If the sync_segment behaved more like a mathematical function, i.e. returned the "result" as one of the return values, it would be clearly visible how it works and would have less side effects. Or we could move the logic to produce the printable results from inside of sync_segment to outside and produce them from the synchronize param we are using for computation (I'd prefer that)